tensorflow/compiler/xla/service/llvm_ir/math_ops.cc

```diff
@@ -14,13 +14,19 @@ limitations under the License.
 ==============================================================================*/
 
 #include "tensorflow/compiler/xla/service/llvm_ir/math_ops.h"
+
 #include "tensorflow/compiler/xla/service/llvm_ir/llvm_util.h"
 
 namespace xla {
 namespace llvm_ir {
 
-llvm::Value* EmitFastTanh(llvm::IRBuilder<>* b, llvm::Value* input) {
+llvm::Value* EmitFastTanh(llvm::IRBuilder<>* b, llvm::Value* input,
+                          const TanhType input_type /* = TanhType::Double*/) {
   llvm::Type* type = input->getType();
+  const float plus_clamp =
+      (input_type == TanhType::Double) ? 9.0 : 7.99881172180175781f;
```

Comment:
Why not derive `input_type` from input->Type() inside the function?

Reply:
```input_type ``` maybe ```vector/array```, so I write a custom enum type.

Reply:
Could we get input type from inside the vector?